File: src/main/java/com/circleescape/server/model/game/Collocation.java
package com.circleescape.server.model.game;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Embeddable
public class Collocation {

    /*@Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(nullable = false)
    private Integer id;*/

    @Column(nullable = false)
    private double duckRadius;//todo embed the duck?
    @Column(nullable = false)
    private double duckAngle;
    @Column(nullable = false)
    private double catAngle;
}

File: src/main/kotlin/com/circleescape/server/model/game/persistence/Collocation.kt
package com.circleescape.server.model.game.persistence

import javax.persistence.Column
import javax.persistence.Embeddable

@Embeddable
class Collocation( //todo embed the duck?
    @Column(nullable = false) var duckRadius: Double = 0.0,
    @Column(nullable = false) var duckAngle: Double = 0.0,
    @Column(nullable = false) var catAngle: Double = 0.0
  /*@Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(nullable = false)
    private Integer id;*/
)